### PR TITLE
check for overage youth, add/show overage error

### PR DESCRIPTION
**Pull request by heyitsmaimai**

### CODE REVIEW

File: src/events/order/cd-event-add-child-ticket.vue [hunk 1]
```diff
@@ -21,6 +21,7 @@
                         :proportions="[2, 2, 3]" data-vv-value-path="value" :data-vv-name="`dob-${id}`" v-validate="'required'"></vue-dob-picker>
       </div>
       <p class="cd-child-ticket__dob-err text-danger" v-show="errors.has(`dob-${id}:required`)">{{ $t('Date of birth is required') }}</p>
+      <p class="cd-child-ticket__dob-u19-err text-danger" v-show="errors.has(`dob-u19-${id}:overage`)">{{ $t('Youths need to be 18 years old or younger.') }}</p>
```
Comment (Wardormeur) [inline src/events/order/cd-event-add-child-ticket.vue]: Update error message
Also needs translation

File: src/events/order/cd-event-add-child-ticket.vue [hunk 2]
```diff
@@ -98,8 +99,19 @@
       applications() {
         OrderStore.commit('setApplications', { id: this.id, applications: this.applications });
       },
+      dob() {
+        if (this.dob && !this.isU19) {
```
Comment (Wardormeur) [inline src/events/order/cd-event-add-child-ticket.vue]: Why check this.dob when this.isU19 checks it as well ?

File: src/events/order/cd-event-add-child-ticket.vue (hunk 1 above)
Comment (heyitsmaimai) [reply src/events/order/cd-event-add-child-ticket.vue]: I think the error msg should be 'Youth is overage (19+)'. Yep I do translations last, that will be done :)

Comment (DanielBrierton) [reply src/events/order/cd-event-add-child-ticket.vue]: Ooohhh... *Over age*. _Overage_ is
> an excess or surplus, especially the amount by which a sum of money is greater than a previous estimate.

File: src/events/order/cd-event-add-child-ticket.vue [hunk 3]
```diff
@@ -18,10 +18,10 @@
         <vue-dob-picker select-class="form-control cd-child-ticket__dob-picker-wrapper-select" v-model="dob"
                         show-labels="false" month-format="short"
                         :placeholders="[$t('Date'), $t('Month'), $t('Year')]"
-                        :proportions="[2, 2, 3]" data-vv-value-path="value" :data-vv-name="`dob-${id}`" v-validate="'required'"></vue-dob-picker>
+                        :proportions="[2, 2, 3]" data-vv-value-path="value" :data-vv-name="`dob-${id}`" v-validate="'required|overage'"></vue-dob-picker>
       </div>
       <p class="cd-child-ticket__dob-err text-danger" v-show="errors.has(`dob-${id}:required`)">{{ $t('Date of birth is required') }}</p>
-      <p class="cd-child-ticket__dob-u19-err text-danger" v-show="errors.has(`dob-u19-${id}:overage`)">{{ $t('Youths need to be 18 years old or younger.') }}</p> 
+      <p class="cd-child-ticket__dob-u18-err text-danger" v-show="errors.has(`dob-${id}:overage`)">{{ $t('Youth tickets are for attendees 18 years or younger. No ticket is required for guardians or parents.') }}</p>
```
Comment (DanielBrierton) [inline src/events/order/cd-event-add-child-ticket.vue]: All the checks are for them being under 18, not 18 and under. The checks are fine, this should be changed to "attendees under 18 years" or "attendees 17 years or younger".

File: src/events/order/cd-event-add-child-ticket.vue [hunk 4]
```diff
@@ -76,6 +76,12 @@
         notes: '',
       };
     },
+    created() {
+      this.$validator.extend('overage', {
+        getMessage: field => `The ${field} value is invalid. Youth is overage (18+)`,
```
Comment (DanielBrierton) [inline src/events/order/cd-event-add-child-ticket.vue]: > Overage
Even after learning that this is a valid spelling for this, I still can't read it as one. 😅 Can we go with "over age" or "over-age"..?

File: src/events/order/cd-event-add-child-ticket.vue (hunk 3 above)
Comment (heyitsmaimai) [reply src/events/order/cd-event-add-child-ticket.vue]: gotcha

File: src/events/order/cd-event-add-child-ticket.vue (hunk 4 above)
Comment (heyitsmaimai) [reply src/events/order/cd-event-add-child-ticket.vue]: yeah i will put the dash

**Pull request merged**